### ISSUE TITLE
curly multi-or-nest should not trigger when starting on the same line instead of the one below

### ISSUE THREAD

**Issue by ThaJay** [eslint/eslint#8831]
**Tell us about your environment**

* **ESLint Version:** v3.19.0
* **Node Version:** v6.10.3
* **npm Version:** 3.10.10

**Please show your full configuration:**

```js
{
  "parser": "babel-eslint",
  "plugins": ["react", "jsx-a11y", "import"],
  "rules": {
    "curly": [error, "multi-or-nest"]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
// Error - Expected { after 'if' condition. (curly)
if (condition) functionCall(
  'with', 'some', parametersOnNewline, and.aLong.getterOfSomeKind(gettingSomething)
)

// correct
if (condition) {
  functionCall(
    'with', 'some', parametersOnNewline, and.aLong.getterOfSomeKind(gettingSomething)
  )
}

// correct
if (condition)
  functionCall('with', 'some', parametersOnNewline, and.aLong.getterOfSomeKind(gettingSomething))
```

**What did you expect to happen?**
I would like the first example to be seen as single line becase it starts on the same line as the `if` statement. The line is too long to have its own line below the if statement (like in the third example), so the first example shows how I like to solve that. This way I use one extra line instead of three. I would still like to use curly braces when there is a nested statement starting on the next line, so that's why I chose `multi-or-nest`.

**What actually happened? Please include the actual, raw output from ESLint.**
even when starting on the same line it flags the code because it either contains a nested statement or spans multiple lines.
first example: `Error - Expected { after 'if' condition. (curly)`

**Comment by kaicataldo** [comment]
According to the docs (and looking at the code), this looks to me to be working as intended. From the [docs](http://eslint.org/docs/rules/curly#multi-or-nest)
> You can use another configuration that forces brace-less if, else if, else, for, while, or do if their body contains only one single-line statement.

Are you proposing an enhancement (this would probably require an addition of an option or more configuration)?

**Comment by ThaJay** [comment]
I am indeed proposing an enhancement. The current output is to be expected but I think it would be nice to be albe to have a braceless multi line statement if it starts on the same line, even with nested enabled.
I would like the following code to pass:
```
if (condition) {
  if (anotherCondition) doThing()
  else doOtherThing()
}

if (condition) doThing(
  withLongParameters.toGetSomething()
)
```
and the following to fail:
```
if (condition)
  if (anotherCondition) doThing()
  else doOtherThing()

if (condition)
  doThing(
    withLongParameters.toGetSomething()
  )
```

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
